### PR TITLE
Build js packages at code-generation time

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Produce Javascript packages without manipulating yarn workspaces when daml2ts is invoked.

### CODE REVIEW

**Comment by shayne-fletcher** [comment]
I've been working on a daml2ts lint test. Unfortunately without success  so far. Fixing that will be in my next PR.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs [hunk 1]
```diff
@@ -193,55 +174,65 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
...
+        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "src" </> "Upgrade.ts")
         assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.js")
-        step "yarn workspaces run lint..."
-        yarnProject ["workspaces", "run", "lint"]
-     ]
+        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.d.ts")
+      step "eslint..."
+      withCurrentDirectory daml2tsDir $ do
+        pkgs <- (\\ ["package.json", "node_modules"]) <$> listDirectory daml2tsDir
+        BSL.writeFile "package.json" $ encode (
```
Comment (shayne-fletcher) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: We should probably lift these constants to the type of the file for easy maintenance as we do in `TsCodeGenMain.hs`.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs [hunk 2]
```diff
@@ -193,55 +174,65 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
...
+        assertFileExists (daml2tsDir </> "davl-0.0.5" </> "lib" </> "DAVL.d.ts")
+        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "src" </> "Upgrade.ts")
         assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.js")
-        step "yarn workspaces run lint..."
-        yarnProject ["workspaces", "run", "lint"]
-     ]
+        assertFileExists (daml2tsDir </> "davl-upgrade-v4-v5-0.0.5" </> "lib" </> "Upgrade.d.ts")
+      step "eslint..."
+      withCurrentDirectory daml2tsDir $ do
+        pkgs <- (\\ ["package.json", "node_modules"]) <$> listDirectory daml2tsDir
```
Comment (shayne-fletcher) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: We should probably lift these constants to the type of the file for easy maintenance as we do in `TsCodeGenMain.hs`.

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs [hunk 3]
```diff
@@ -31,33 +31,43 @@ import Test.Main
...
 
 main :: IO ()
-main =
-    withTempDir $ \tmpDir -> do
+main = do
+    setEnv "TASTY_NUM_THREADS" "1" True
     -- We manipulate global state via the working directory and
     -- the environment so running tests in parallel will cause trouble.
-    setEnv "TASTY_NUM_THREADS" "1" True
+    yarn : damlTypesPath : args <- getArgs
```
Comment (shayne-fletcher) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs]: Unfortunately, doing this is the only way I've found to reliably get at the directory containing yarn.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs [hunk 4]
```diff
@@ -15,18 +15,28 @@ import qualified DA.Daml.LF.Ast.Version as LF
...
     damlc <- locateRunfiles (mainWorkspace </> "compiler" </> "damlc" </> exe "damlc")
     daml2ts <- locateRunfiles (mainWorkspace </> "language-support" </> "ts" </> "codegen" </> exe "daml2ts")
-    davl <- locateRunfiles ("davl" </> "released")
-    yarnPath : damlTypesPath : args <- getArgs
     yarn <- locateRunfiles (mainWorkspace </> yarnPath)
-    damlTypes <- (</> damlTypesPath) <$> getCurrentDirectory
+    damlTypes <- locateRunfiles (mainWorkspace </> damlTypesPath)
+    davl <- locateRunfiles ("davl" </> "released")
+    -- TODO (SF,2020-03-24): Factor out 'withEnv' from
+    -- 'DA/DamlAssistant/Tests.hs' into a library function and use it here.
```
Comment (shayne-fletcher) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Note this.

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs [hunk 5]
```diff
@@ -31,33 +31,43 @@ import Test.Main
...
+import DA.Directory
 import DA.Bazel.Runfiles
 import DA.Daml.Helper.Run
 import SdkVersion
 
 main :: IO ()
-main =
-    withTempDir $ \tmpDir -> do
+main = do
+    setEnv "TASTY_NUM_THREADS" "1" True
```
Comment (hurryabit) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs]: Why did you move this above the explaining comment?

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs (hunk 3 above)
Comment (hurryabit) [reply daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs]: That's fine. We're doing this in other places too.

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs [hunk 6]
```diff
@@ -490,8 +503,22 @@ codegenTests codegenDir = testGroup "daml codegen"
...
                     withCurrentDirectory projectDir $ do
                         callCommandQuiet "daml build"
-                        let darFile = projectDir</> ".daml/dist/proj-" ++ lang ++ "-0.0.1.dar"
-                            outDir  = projectDir</> "generated" </> lang
+                        let darFile = projectDir </> ".daml/dist/proj-" ++ lang ++ "-0.0.1.dar"
+                            outDir  = projectDir </> "generated" </> lang
+                        when (lang == "ts") $ do
+                          createDirectoryIfMissing True "generated"
+                          withCurrentDirectory "generated" $ do
+                            -- SDK version is 0.0.0; daml2ts needs
```
Comment (hurryabit) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs]: Isn't that how it looks like?
```suggestion
                            -- SDK version is 0.0.0-SDKVERSION; daml2ts needs
```

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs [hunk 7]
```diff
@@ -490,8 +503,22 @@ codegenTests codegenDir = testGroup "daml codegen"
...
+                            -- 'daml-types' to be here in the filesystem...
+                            copyDirectory damlTypes "daml-types"
+                            BSL.writeFile "package.json" $ encode (
+                            -- ... and this package.json so it can find it.
+                              object
+                                [ "private" .= True
+                                , "workspaces" .= [T.pack lang]
+                                , "resolutions" .= HMS.fromList ([("@daml/types", "file:daml-types")] :: [(T.Text, T.Text)])
+                                ]
+                              )
```
Comment (hurryabit) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs]: Why not spend more 💵s?
```suggestion
                            BSL.writeFile "package.json" $ encode $
                            -- ... and this package.json so it can find it.
                              object
                                [ "private" .= True
                                , "workspaces" .= [T.pack lang]
                                , "resolutions" .= HMS.fromList ([("@daml/types", "file:daml-types")] :: [(T.Text, T.Text)])
                                ]
```

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs [hunk 8]
```diff
@@ -490,8 +503,22 @@ codegenTests codegenDir = testGroup "daml codegen"
                     callCommandQuiet $ unwords ["daml new", projectDir, "skeleton"]
                     withCurrentDirectory projectDir $ do
                         callCommandQuiet "daml build"
-                        let darFile = projectDir</> ".daml/dist/proj-" ++ lang ++ "-0.0.1.dar"
-                            outDir  = projectDir</> "generated" </> lang
+                        let darFile = projectDir </> ".daml/dist/proj-" ++ lang ++ "-0.0.1.dar"
+                            outDir  = projectDir </> "generated" </> lang
+                        when (lang == "ts") $ do
```
Comment (hurryabit) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs]: Can you please leave _one_ comment that explains what we're doing here, namely making `@daml-types-0.0.9-SDKVERSION` available to yarn, instead of having multiple comments spread out below.

Other than that, very nice solution!

File: docs/source/daml2ts/index.rst [hunk 9]
```diff
@@ -21,7 +21,7 @@ Usage
 
 ``daml2ts`` is invoked via the DAML SDK assistant.
 
-In outline, the command to generate TypeScript from DAML is ``daml codegen ts DAR -o OUTDIR –main-package-name=PACKAGE`` where ``DAR`` is the path to a DAR file (generated via ``daml build``), ``OUTDIR`` is a directory where you want the TypeScript to be written and ``PACKAGE`` is a desired TypeScript package name.
+In outline, the command to generate TypeScript from DAML is ``daml codegen ts DAR -o OUTDIR`` where ``DAR`` is the path to a DAR file (generated via ``daml build``), ``OUTDIR`` is a directory where you want the TypeScript to be written and ``PACKAGE`` is a desired TypeScript package name.
```
Comment (hurryabit) [inline docs/source/daml2ts/index.rst]: There is no more `PACKAGE` parameter and we're finally writing JavaScript + typing. We should embrace that:
```suggestion
In outline, the command to generate JavaScript and TypeScript typings from DAML is ``daml codegen ts DAR -o OUTDIR`` where ``DAR`` is the path to a DAR file (generated via ``daml build``) and ``OUTDIR`` is a directory where you want the JavaScript to be written.
```
Can you please check there are no more references to `PACKAGE` elsewhere.

File: docs/source/daml2ts/index.rst [hunk 10]
```diff
@@ -31,15 +31,15 @@ Here's a complete example that generates TypeScript from a project produced from
...
 - On execution of these commands:
 
-  - The directory ``my-proj/generated/ts`` contains the generated TypeScript source files;
+  - The directory ``my-proj/daml2ts`` contains generated TypeScript and Javascript artifacts;
   - The files are arranged into directories;
-  - One of those directories will be named as per the ``PACKAGE`` argument and will contain the TypeScript definitions corresponding to the DAML files in the project;
-  - For example, ``generated/ts/my-proj/Main.ts`` contains the TypeScript definitions for ``daml/Main.daml``;
-  - The remaining directories contain supporting TypeScript corresponding to modules of the DAML standard library;
+  - One of those directories will be named as per the ``PACKAGE`` argument and will contain the definitions corresponding to the DAML files in the project;
+  - For example, ``generated/ts/my-proj/Main.ts`` contains the definitions for ``daml/Main.daml``;
```
Comment (hurryabit) [inline docs/source/daml2ts/index.rst]: Isn't there a `src` somewhere in the path to `Main.ts` as well?

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 11]
```diff
@@ -80,7 +78,7 @@ newtype Script = Script {unScript :: T.Text}
 data Options = Options
     { optInputDars :: [FilePath]
     , optOutputDir :: FilePath
-    , optInputPackageJson :: FilePath
+    , optInputPackageJson :: Maybe FilePath -- Deprecated.
```
Comment (hurryabit) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: Can you please create an issue to make sure we remove this flag after the next release.

File: language-support/ts/codegen/tests/build-and-lint.sh [hunk 12]
```diff
@@ -57,11 +57,14 @@ cp -rL $DAML_LEDGER/* $TMP_DAML_LEDGER
...
-$YARN workspaces run build
-$YARN workspaces run lint
+# Call daml2ts.
+PATH=`dirname $YARN`:$PATH $DAML2TS -o daml2ts $DAR
 
+# The previous step provides all the daml2ts Javascript needed by the
+# build-and-lint-test workspace.
+$YARN install --pure-lockfile > /dev/null 2>&1
+$YARN workspaces run build  # Build it.
+$YARN workspaces run lint   # No great value in this but nonetheless, lint it.
```
Comment (hurryabit) [inline language-support/ts/codegen/tests/build-and-lint.sh]: Shouldn't we be able to do this without yarn workspaces now? It would be great to confirm that in this test.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs [hunk 13]
```diff
@@ -133,47 +143,18 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
         buildProject []
       withCurrentDirectory here $ do
         step "daml2ts..."
-        setupWorkspace
-        (exitCode, _, err) <- readProcessWithExitCode daml2ts ([groverDar, superGroverDar] ++ ["-o", daml2tsDir, "-p", here </> "package.json"]) ""
+        copyDirectory damlTypes "daml-types"
+        writePackageJson
```
Comment (hurryabit) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Why not make
```haskell
setupWorkspace = do
  copyDirectory
  writePackageJson
```
Maybe under the name `setupYarnEnvironment` or something similar?

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs [hunk 14]
```diff
@@ -193,55 +174,65 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
...
-        daml2tsProject [groverDar, groverDar] daml2tsDir (here </> "package.json")
+        writePackageJson
+        copyDirectory damlTypes "daml-types"
+        daml2tsProject [groverDar, groverDar] daml2tsDir
         assertFileExists (groverTsSrc </> "Grover.ts")
+        assertFileExists (groverTsLib </> "Grover.js")
+        assertFileExists (groverTsLib </> "Grover.d.ts")
         assertFileExists (groverTsSrc </> "packageId.ts")
+        assertFileExists (groverTsLib </> "packageId.js")
+        assertFileExists (groverTsLib </> "packageId.d.ts")
```
Comment (hurryabit) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: We should also check for `index.*`. It might make sense to have a function like
```haskell
assertTsFileExists dir file = do
  assertFileExists (dir </> "src" </> file <.> "ts")
  assertFileExists (dir </> "lib" </> file <.> "js")
  assertFileExists (dir </> "lib" </> file <.> "d.ts")
```
and call
```haskell
mapM_ (assertTsFileExists groverTs) ["index", "Grover", "packageId"]
```
That would be beneficial below as well.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 2 above)
Comment (hurryabit) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: I agree.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs [hunk 15]
```diff
@@ -278,5 +256,14 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
...
         ["build-options: [--target=" <> LF.renderVersion ver <> "]" | Just ver <- [mbLfVersion]]
 
+    writePackageJson :: IO ()
+    writePackageJson = BSL.writeFile "package.json" $ encode packageJson
+      where
+        packageJson = object
+          [ "private" .= True
+          , "workspaces" .= (["daml2ts"] :: [T.Text])
+          , "resolutions" .= HMS.fromList ([("@daml/types", "file:daml-types")] :: [(T.Text, T.Text)])
+          ]
```
Comment (hurryabit) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: We have the functionality of setting up the yarn environment for tests twice now. Here and way further up. We should factor that out, probably in a followup PR.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs [hunk 16]
```diff
@@ -193,55 +174,65 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
...
+              ([ ("@daml/types", "file:../daml-types")
+               , ("@mojotech/json-type-validation", "^3.1.0")
+               ] :: [(T.Text, T.Text)])
+            , "workspaces" .= pkgs
+            , "name" .= ("daml2ts" :: T.Text)
+            , "version" .= ("0.0.0" :: T.Text)
+            ])
+        callProcessSilent yarn ["install", "--pure-lockfile"]
+        callProcessSilent yarn ["workspaces", "run", "lint"]
+  ]
```
Comment (hurryabit) [inline language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: I have a suspicion that this only works because we write too much in the `package.json` file for every package. I'm fine with keeping it in there for now and remove it in a followup PR.

File: daml-assistant/integration-tests/BUILD.bazel [hunk 17]
```diff
@@ -71,14 +71,19 @@ da_haskell_test(
     name = "integration-tests",
     timeout = "long",
     srcs = glob(["src/**/*.hs"]),
+    args = [
+        "$(location //:yarn)",
+        "daml-types-not-available" if is_windows else "$(location //language-support/ts/daml-types:npm_package)",
```
Comment (hurryabit) [inline daml-assistant/integration-tests/BUILD.bazel]: We should leave a comment here that this test is not run in Windows but we need to pass something to make the test harness happy.

**Comment by hurryabit** [comment]
Two notes, so we don't forget it:
- Let's remove the `"scripts"` section from the `package.json` generated for each package.
- Let's not generate the `.eslintrc.json` files anymore. (Let's produce them somehow in our tests instead.)

File: daml-assistant/integration-tests/BUILD.bazel (hunk 17 above)
Comment (shayne-fletcher) [reply daml-assistant/integration-tests/BUILD.bazel]: Ok

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs (hunk 5 above)
Comment (shayne-fletcher) [reply daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs]: Actually, I introduced the explaining comment 😄 I'll move the comment above the action.

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs (hunk 8 above)
Comment (shayne-fletcher) [reply daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs]: Ok

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs (hunk 7 above)
Comment (shayne-fletcher) [reply daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs]: Lol :) Ok

File: docs/source/daml2ts/index.rst (hunk 9 above)
Comment (shayne-fletcher) [reply docs/source/daml2ts/index.rst]: (1) Fixed.
(2) Done.

File: docs/source/daml2ts/index.rst (hunk 10 above)
Comment (shayne-fletcher) [reply docs/source/daml2ts/index.rst]: Yes. Fixed.

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 11 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: https://github.com/digital-asset/daml/issues/5194

File: language-support/ts/codegen/tests/build-and-lint.sh (hunk 12 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/build-and-lint.sh]: Done.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 4 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Tracking in issue https://github.com/digital-asset/daml/issues/5196

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 13 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Ok

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 14 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: This is a very nice idea. Done.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 2 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Done.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 16 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: I hope your suspicions are unfounded.
```haskell

    packageJson :: Value
    packageJson = object
      [ "name" .= ("daml2ts" :: T.Text)
      , "version" .= version
      , "dependencies" .= pkgDependencies config
      , "devDependencies" .= pkgDevDependencies config
      ]
```
where,
```haskell
configConsts :: T.Text -> ConfigConsts
configConsts sdkVersion = ConfigConsts
  { pkgDependencies = HMS.fromList
      [ (NpmPackageName "@mojotech/json-type-validation", NpmPackageVersion "^3.1.0")
      , (NpmPackageName "@daml/types", NpmPackageVersion sdkVersion)
      ]
  , pkgDevDependencies = HMS.fromList
      [ (NpmPackageName "typescript", NpmPackageVersion "~3.7.3")
      ]
  , pkgScripts = HMS.fromList
      [ (ScriptName "build", Script "tsc")
      , (ScriptName "lint", Script "eslint --ext .ts --max-warnings 0 src/")
      ]
  }
```
Let me know!

Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: I do observe that the `pkgDependencies` are not required for the lint test. Removing those in this PR.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 15 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Well yes, we have two occasions to write json:
(1)
```haskell
        BSL.writeFile "package.json" $ encode (
          object
            [ "private" .= True
            , "devDependencies" .= pkgDevDependencies configConsts
            , "workspaces" .= pkgs
            , "name" .= ("daml2ts" :: T.Text)
            , "version" .= ("0.0.0" :: T.Text)
            ])
```
(2)
```haskell
    writePackageJson :: IO ()
    writePackageJson = BSL.writeFile "package.json" $ encode packageJson
      where
        packageJson = object
          [ "private" .= True
          , "workspaces" .= (["daml2ts"] :: [T.Text])
          , "resolutions" .= HMS.fromList ([("@daml/types", "file:daml-types")] :: [(T.Text, T.Text)])
          ]
```
Although similar, they do have very different contents. Do you still feel this deserves refactoring?

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 16 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Oh wait! I think you mean we do something similar in the DAML assistant too. Yes, we should definitely factor this out then. I'll put it on the stack with the 2 notes below.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 15 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Oh wait! I think you mean we do something similar in the DAML assistant too. Yes, we should definitely factor this out then. I'll put it on the stack with the 2 notes below.

**Comment by shayne-fletcher** [comment]
Follow up in https://github.com/digital-asset/daml/pull/5201. There are 4 things we know we want to do remaining:

* Let's remove the `"scripts"` section from the `package.json` generated for each package.
* Let's not generate the `.eslintrc.json` files anymore. (Let's produce them somehow in our tests instead.)
* Reuse `writePackageJson` between the daml2ts test and the daml assistant integration test
* Reuse `withEnv` from the daml assistant tests in the daml2ts test.

**Comment by shayne-fletcher** [comment]
The last of those points is tracked in https://github.com/digital-asset/daml/issues/5196.

File: language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs (hunk 16 above)
Comment (hurryabit) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Does that mean we can delete `pkgScripts`? Let's do that then! 🔥

Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs]: Done.
